Use cp -pPR instead of cp -a to fix install on Mac OS X Leopard and earlier.
https://github.com/redis/hiredis/pull/596
https://github.com/redis/hiredis/commit/0e9af8acb6bbd5b77525f17883fad1a24fd828bd
--- Makefile.orig	2015-09-16 05:25:02.000000000 -0500
+++ Makefile	2018-05-14 17:56:18.000000000 -0500
@@ -58,7 +58,6 @@
 ifeq ($(uname_S),SunOS)
   REAL_LDFLAGS+= -ldl -lnsl -lsocket
   DYLIB_MAKE_CMD=$(CC) -G -o $(DYLIBNAME) -h $(DYLIB_MINOR_NAME) $(LDFLAGS)
-  INSTALL= cp -r
 endif
 ifeq ($(uname_S),Darwin)
   DYLIBSUFFIX=dylib
@@ -161,11 +160,7 @@
 dep:
 	$(CC) -MM *.c
 
-ifeq ($(uname_S),SunOS)
-  INSTALL?= cp -r
-endif
-
-INSTALL?= cp -a
+INSTALL?= cp -pPR
 
 $(PKGCONFNAME): hiredis.h
 	@echo "Generating $@ for pkgconfig..."
